ODIN_II/SRC/ast_elaborate.cpp

```diff
@@ -2216,36 +2216,43 @@ void update_instance_parameter_table(ast_node_t* instance, STRING_CACHE* instanc
                               "Can't find parameter name %s in module %s\n",
                               param_id, instance->children[0]->types.identifier);
             } else {
-                /* update this parameter and remove all other overrides with this name */
-                free_whole_tree(((ast_node_t*)instance_param_table_sc->data[sc_spot]));
-                instance_param_table_sc->data[sc_spot] = (void*)parameter_override_list->children[j]->children[5];
-                parameter_override_list->children[j]->children[5] = NULL;
-
-                for (int k = 0; k < j; k++) {
-                    if (!strcmp(parameter_override_list->children[k]->children[0]->types.identifier, param_id)) {
-                        /**
+                ast_node_t* temp = (ast_node_t*)instance_param_table_sc->data[sc_spot];
+                if (temp->types.variable.is_parameter) {
+                    /* update this parameter and remove all other overrides with this name */
+                    free_whole_tree(((ast_node_t*)instance_param_table_sc->data[sc_spot]));
+                    instance_param_table_sc->data[sc_spot] = (void*)parameter_override_list->children[j]->children[5];
+                    parameter_override_list->children[j]->children[5] = NULL;
+
+                    for (int k = 0; k < j; k++) {
+                        if (!strcmp(parameter_override_list->children[k]->children[0]->types.identifier, param_id)) {
+                            /**
                          * TODO: is this needed?
                          * ast_node_t* temp = parameter_override_list->children[k]; 
                          * */
-                        remove_child_from_node_at_index(parameter_override_list, k);
-                        //temp = NULL;
-                        j--;
-                        k--;
+                            remove_child_from_node_at_index(parameter_override_list, k);
+                            //temp = NULL;
+                            j--;
+                            k--;
+                        }
                     }
-                }
-                /**
+                    /**
                  * TODO: is this needed?
                  * ast_node_t* temp = parameter_override_list->children[j];
                  * */
-                remove_child_from_node_at_index(parameter_override_list, j);
-                //temp = NULL;
+                    remove_child_from_node_at_index(parameter_override_list, j);
+                    //temp = NULL;
+                } else if (temp->types.variable.is_localparam) {
+                    warning_message(AST, parameter_override_list->loc,
```

Comment:
you should still free everything related, the only thing that you should guard  is swapping the node in the table

Reply:
this would apply to the error thrown above also. but more importantly, can you answer the TODO?